Library/Homebrew/cmd/uninstall.rb

```diff
@@ -6,44 +6,58 @@
 
 require "keg"
 require "formula"
+require "diagnostic"
 require "migrator"
 
 module Homebrew
   def uninstall
     raise KegUnspecifiedError if ARGV.named.empty?
 
-    if !ARGV.force?
-      ARGV.kegs.each do |keg|
-        keg.lock do
-          puts "Uninstalling #{keg}... (#{keg.abv})"
-          keg.unlink
-          keg.uninstall
-          rack = keg.rack
-          rm_pin rack
-
-          if rack.directory?
-            versions = rack.subdirs.map(&:basename)
-            verb = versions.length == 1 ? "is" : "are"
-            puts "#{keg.name} #{versions.join(", ")} #{verb} still installed."
-            puts "Remove all versions with `brew uninstall --force #{keg.name}`."
-          end
-        end
-      end
-    else
-      ARGV.named.each do |name|
+    kegs_by_rack = if ARGV.force?
+      Hash[ARGV.named.map { |name|
```

Comment:
Yes, using braces for functional blocks is nicer for method chaining.

An example from `caveats.rb`:

```ruby
    @keg ||= [f.prefix, f.opt_prefix, f.linked_keg].map do |d|
      begin
        Keg.new(d.resolved_path)
      rescue
        nil
      end
    end.compact.first
```

vs.

```ruby
    @keg ||= [f.prefix, f.opt_prefix, f.linked_keg].map { |d|
      begin
        Keg.new(d.resolved_path)
      rescue
        nil
      end
    }.compact.first
```